sklearn/decomposition/_pca.py

```diff
@@ -360,6 +367,7 @@ def __init__(
         self.svd_solver = svd_solver
         self.tol = tol
         self.iterated_power = iterated_power
+        self.n_oversamples_rate = n_oversamples_rate
         self.random_state = random_state
 
     def fit(self, X, y=None):
```

Comment:
Can you please add a scalar validation for `n_oversamples_rate` using `check_scalar` at the start of `PCA.fit` and test it, similarly to what has been done in https://github.com/scikit-learn/scikit-learn/pull/20723/files?